chain-impl-mockchain/src/lib.rs

```diff
@@ -30,5 +30,13 @@ pub mod update;
 pub mod utxo;
 pub mod value;
 
-#[cfg(test)]
-mod tests {}
+#[macro_use]
+extern crate cfg_if;
+
+cfg_if! {
+   if #[cfg(test)] {
+        pub mod testing;
```

Comment:
I wonder why doesn't this work for in-tree test builds.

Reply:
There is an open issue about this: rust-lang/cargo#2911

A workaround is to also only enable tests that depend on the property api if the `property-test-api` feature is enabled (or maybe it's how it should be, since the property test API is an optional part of the crate and should be tested as such.) Then in CI, we can use the feature flag.

Reply:
On a better thought, we should not bother with test feature flags before looking into what may be the root of the problem here: wrong placement of test code. Any test code that depends on the property test impls in the source tree of the crate should belong in _unit_ tests (in `#[cfg(test)] mod tests { ... }`). Any generally useful test utilities that look like they might be used from other places than the in-crate tests should be moved into `mod testing` and made public.